### ISSUE TITLE
Namelist.__init__ does not convert list of dicts to list of Namelists

### ISSUE THREAD

**Issue by gitporst** [marshallward/f90nml#61]
in line 40 in namelist.py
```
        # Convert any internal dicts to Namelists
        for key, val in self.items():
            if isinstance(val, dict):
                self[key] = Namelist(val)

```
there is a bug as there also needs to be a check for list of dicts so we can parse FORTRAN vectors of derived types correctly.

**Comment by marshallward** [comment]
Yes, I can reproduce this:

    >>> x = {'x_nml': {'a': [{'b': 1, 'c': 2}, {'b': 3, 'c': 4}]}}
    >>> n = f90nml.Namelist(x)
    >>> print(n)

I'll try to fix this today.

**Comment by marshallward** [comment]
This should now be fixed in the latest commit: 53775732164bc8fb768d4025a12994d8539ac13a

From the commit log


> The Namelist constructor was not converting lists of dicts into list of Namelists, which was causing various problems related to indexing and expectations of the start_index property.
> 
> There also was not very good support for arrays of dicts without a start index.  This was presumed to not be a problem, since arrays of derived types require an explicit index, but it was an issue when initialising with a Python dict.

Could you try this for me?

**Comment by gitporst** [comment]
thank you, this issue is now fixed, lists of dicts are correctly converted to lists of Namelists.
